### ISSUE TITLE
Multiple masks cause confused output

### ISSUE THREAD

**Issue by darshats** [lllyasviel/ControlNet#370]
Hi,
Thanks for providing this amazing capability for SD!

I had a question for multiple objects placed in the image. Sometimes I'm finding if two masks are provided for the image inpainting (via canny or segments) it causes confused output. For example, a canny outline for an apple and a canny outline for a succulent plant is given. Once inpainting is done it will at times try to squeeze the green plant into the apple. At times it may not draw the second object at all.

Was training done with multiple objects provided? And is there a way to better achieve this?
Thanks!

**Comment by Poopoopeepee42069** [comment]
I’ve noticed this as well.

**Comment by 2a3b4c** [comment]
have you solve this problem? @darshats

**Comment by geroldmeisinger** [comment]
> Was training done with multiple objects provided?

I don't think so as Stable Diffusion doesn't recognize "spatial" keywords anyway ("apple at upper left corner") and suffers from concept bleeding ("red apple and green plant" doesn't work, see "regional prompting").

> And is there a way to better achieve this?

do a multi step approach: inpaint first object, inpaint second object
